src/main/java/ch/njol/skript/events/EvtExplode.java

```diff
@@ -0,0 +1,177 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.effects.EffExplosion;
+import ch.njol.skript.entity.EntityType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Color;
+import ch.njol.skript.util.ColorRGB;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.FireworkEffect;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.bukkit.event.entity.FireworkExplodeEvent;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EvtExplode extends SkriptEvent {
```

Comment:
I don't know about this. If none of the three versions share code, why combine them? It just makes one harder to understand class.